scripts/globals/moghouse.lua

```diff
@@ -169,6 +169,25 @@ xi.moghouse.moogleTrade = function(player, npc, trade)
                 player:messageSpecial(zones[player:getZoneID()].text.MOG_LOCKER_OFFSET + 2, xi.moghouse.getMogLockerExpiryTimestamp(player))
             end
         end
+
+        local eggComponents =
+        {
+            xi.items.EGG_LOCKER,
+            xi.items.EGG_TABLE,
+            xi.items.EGG_STOOL,
+            xi.items.EGG_LANTERN,
+        }
+
+        if npcUtil.tradeHasExactly(trade, eggComponents) then
```

Comment:
tradeHasExactly functions use confirmItem, while tradeComplete() consumes the entire trade inventory.  This should be changed to player:confirmTrade()